### PR TITLE
feat(aws): Add new kafka service

### LINKED ISSUE #3049: support for aws msk
### New feature motivation

enable support for aws msk (Managed service for kafka)

### Solution Proposed

Enable msk service support in prowler

### Describe alternatives you've considered

Not Yet

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puchy22**
### Context

Add new Kafka service with one basic check.

### Description

It adds new service code, basic check code to ensure using the latest version of Kafka and their respective unit tests.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "Severity": "medium",
+  "ResourceType": "",
+  "Description": "Ensure that your Amazon Managed Streaming for Apache Kafka (MSK) cluster is using the latest version to benefit from the latest security features, bug fixes, and performance improvements.",
+  "Risk": "Running an outdated version of Amazon MSK may expose your cluster to security vulnerabilities, bugs, and performance issues.",
+  "RelatedUrl": "https://docs.aws.amazon.com/lightsail/latest/userguide/amazon-lightsail-databases.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws kafka update-cluster-configuration --cluster-arn <arn_cluster> --current-version <current_version> --target-version <latest_version>",
+      "NativeIaC": "",
+      "Other": "",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json]:
```suggestion
      "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/MSK/enable-apache-kafka-latest-security-features.html",
```

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Change the database to private mode.",
+      "Url": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/MSK/enable-apache-kafka-latest-security-features.html"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json]:
```suggestion
      "Url": "https://docs.aws.amazon.com/msk/latest/developerguide/version-support.html#version-upgrades"
```

File: prowler/providers/aws/services/kafka/kafka_service.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        self.kafka_versions = []
+        self.__threading_call__(self.__list_kafka_versions__)
+
+    def __list_clusters__(self, regional_client):
+        try:
+            cluster_paginator = regional_client.get_paginator("list_clusters")
+
+            for page in cluster_paginator.paginate():
+                for cluster in page["ClusterInfoList"]:
+                    self.clusters[cluster.get("ClusterArn", "")] = Cluster(
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_service.py]: Please use the resource filter:
```
                    if not self.audit_resources or (
                        is_resource_filtered(arn, self.audit_resources)
                    ):
```

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            report.resource_arn = arn_cluster
+            report.resource_tags = cluster.tags
+            report.status = "PASS"
+            report.status_extended = (
+                f"Kafka cluster '{cluster.name}' is using the latest version."
+            )
+
+            if cluster.kafka_version != getattr(
+                kafka_client.kafka_versions[-1], "version", ""
+            ):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.py]: `getattr` is not needed here ;)

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "kafka_cluster_use_latest_version",
```
Comment (sergargar) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json:3]:
```suggestion
  "CheckID": "kafka_cluster_use_latest_security_features",
```

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  "RelatedUrl": "https://docs.aws.amazon.com/lightsail/latest/userguide/amazon-lightsail-databases.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws kafka update-cluster-configuration --cluster-arn <arn_cluster> --current-version <current_version> --target-version <latest_version>",
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/MSK/enable-apache-kafka-latest-security-features.html",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Change the database to private mode.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json]: Please, verify this.

File: prowler/providers/aws/services/kafka/kafka_service.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                    arn = cluster.get(
+                        "ClusterArn",
+                        f"{self.account_arn_template}/{cluster.get('ClusterName', '')}",
+                    )
+
+                    if not self.audit_resources or is_resource_filtered(
+                        arn, self.audit_resources
+                    ):
+                        self.clusters[cluster.get("ClusterArn", "")] = Cluster(
+                            id=cluster["ClusterArn"].split(":")[-1].split("/")[-1],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_service.py]:
```suggestion
                            id=arn.split(":")[-1].split("/")[-1],
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments!

File: prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json (hunk 5 above)
Comment (MrCloudSec) [inline prowler/providers/aws/services/kafka/kafka_cluster_use_latest_version/kafka_cluster_use_latest_version.metadata.json]:
```suggestion
  "CheckID": "kafka_cluster_uses_latest_version",
```

**Review by MrCloudSec — APPROVED** [review]
Good job!!

**Pull request merged**